src/sagemaker_xgboost_container/algorithm_mode/serve_utils.py

```diff
@@ -110,3 +163,269 @@ def predict(booster, model_format, dtest, input_content_type):
     return booster.predict(dtest,
                            ntree_limit=getattr(booster, "best_ntree_limit", 0),
                            validate_features=False)
+
+
+def is_selectable_inference_output():
+    return SAGEMAKER_INFERENCE_OUTPUT in os.environ
+
+
+def get_selected_output_keys():
+    """Get the selected output content keys from the `SAGEMAKER_INFERENCE_OUTPUT` env var.
+
+    :return: selected output content keys (list of str)
+    """
+    if is_selectable_inference_output():
+        return os.getenv(SAGEMAKER_INFERENCE_OUTPUT).replace(' ', '').lower().split(',')
+    raise RuntimeError("'SAGEMAKER_INFERENCE_OUTPUT' environment variable is not present. "
+                       "Selectable inference content is not enabled.")
+
+
+def _get_labels(objective, num_class=""):
+    """Get the labels for a classification problem.
+
+    :param objective: xgboost objective function (str)
+    :param num_class: number of classes in multiclass (str, optional)
+    :return: classes (list of integers) or np.nan
+    """
+    if "binary:" in objective:
+        return [0, 1]
+    if "multi:" in objective and num_class:
+        return list(range(int(num_class)))
+    return np.nan
+
+
+def _get_predicted_label(objective, raw_prediction):
+    """Get the predicted label for a classification problem.
+
+    :param objective: xgboost objective function (str)
+    :param raw_prediction: xgboost predict output (numpy array or numpy primitive)
+    :return: predicted label (int or float) or np.nan
+    """
+    if objective in [BINARY_HINGE, MULTI_SOFTMAX]:
+        return raw_prediction.item()
+    if objective in [BINARY_LOG]:
+        return int(raw_prediction > 0.5)
+    if objective in [BINARY_LOGRAW]:
+        return int(raw_prediction > 0)
+    if objective in [MULTI_SOFTPROB]:
+        return np.argmax(raw_prediction).item()
+    return np.nan
+
+
+def _get_probability(objective, raw_prediction):
+    """Get the probability score for a classification problem.
+
+    A probability score is a value between 0.0 and 1.0. In binary classification,
```

Comment:
Just leave the critical info here, which is that this is the prob of the winning class.